### DISCUSSION TITLE
Administrator Locked 2FA - Docker M1 Silicon

### DISCUSSION THREAD

**Discussion by staminna** [directus/directus#17456] [Archive]
### Describe the Bug

I only have a single user and it is the Admin with 2FA enabled.

I don't know how to use npx directus cli to generate a UUID for a new Admin user.
I cannot make API calls to regenerate a new token because I don't know the 2FA token or because I lost the Bearer Token or I am not making the POST request to the API correctly.

### To Reproduce

Lose or don't make note of the Bearer token, eventually with time the 2FA gets out of sync and Admin becomes locked.
Invalid 2-step verification code, even though I am sure it is synced with Google Authenticator for Android 12 because I followed the steps: 
<img width="753" alt="google-top" src="https://user-images.githubusercontent.com/515180/217628446-a74138b7-540e-4762-91ba-2ea8c331e453.png">

### Errors Shown

I think I am locked. If you can help with Postman I will make a report with prints but I cannot get a new Bearer token for new admin without 2FA working. 

### What version of Directus are you using?

9.22.3

### What version of Node.js are you using?

v18.13.0

### What database are you using?

Postgres  13

### What browser are you using?

Brave

### How are you deploying Directus?

Docker locally

**Comment by staminna** [comment]
I was able to use ChatGPT to get the help I couldn't from Discord. I was locked up for 4 weeks

By querying the database for the "Administrator" UUID I was able to create a new Administrator and then disable 2FA.
Lesson learned I regenerated and kept the Tokens but the 2FA should have 100%  tested coverage if it doesn't already have it.

**Reply by rijkvanzanten** [reply]
> I was able to use ChatGPT to get the help I couldn't from Discord. I was locked up for 4 weeks

The Discord help chat is a community driven Q&A channel. There's no guarantees on answers or times. If you're having trouble in hosting, maintenance, or support, consider using [Directus Cloud](https://directus.cloud/) instead.

**Comment by paescuj** [comment]
Seems like you've found a solution yourself in the meantime, so I'm just commenting here for completion.
The CLI can be used to create a new admin role and user:
```bash
npx directus roles create --role <role-name> --admin true

npx directus users create --email <user-email> --password <password> --role <role-uuid>
```

This is documented over here: https://docs.directus.io/self-hosted/cli.html#creating-users

**Comment by rijkvanzanten** [comment]
> eventually with time the 2FA gets out of sync and Admin becomes locked.

This doesn't really make any sense to me. 2FA is a single fixed secret token that's stored on the device and in the database, so that doesn't change over time. The one thing that does matter however is that the server time and device time are accurate and in sync. In this case, I'm pretty certain that you're trouble was caused by the clock being misconfigured on either your mobile device, or the server.

**Comment by staminna** [comment]
The clocks have always been synced automatically by MacOS and Android and have worked dozens of times in Directus and in several services, such as Github, Instagram, Dropbox, Google, and others. I did not travel or changed the timezone.

**Reply by paescuj** [reply]
More importantly the server clock must be accurate (by using NTP...).
